pkg/proxy/winkernel/hcnutils.go

```diff
@@ -346,13 +376,157 @@ func (hns hns) getLoadBalancer(endpoints []endpointsInfo, flags loadBalancerFlag
+	return loadbalancer.Create()
+}
+
+func (h *RealHCN) DeleteLoadBalancer(loadbalancer *hcn.HostComputeLoadBalancer) error {
+	return loadbalancer.Delete()
+}
+
+func (h *RealHCN) DeleteEndpoint(endpoint *hcn.HostComputeEndpoint) error {
+	return endpoint.Delete()
+}
+
+// Here are some helper methods for proxier.go
+
+func getNetworkName(hnsNetworkName string) (string, error) {
+	if len(hnsNetworkName) == 0 {
+		klog.V(3).InfoS("Flag --network-name not set, checking environment variable")
+		hnsNetworkName = os.Getenv("KUBE_NETWORK")
+		if len(hnsNetworkName) == 0 {
+			return "", fmt.Errorf("Environment variable KUBE_NETWORK and network-flag not initialized")
+		}
+	}
+	return hnsNetworkName, nil
+}
+
+func getNetworkInfo(hns IHCNUtils, hnsNetworkName string) (*hnsNetworkInfo, error) {
+	hnsNetworkInfo, err := hns.getNetworkByName(hnsNetworkName)
+	for err != nil {
+		klog.ErrorS(err, "Unable to find HNS Network specified, please check network name and CNI deployment", "hnsNetworkName", hnsNetworkName)
+		time.Sleep(1 * time.Second)
+		hnsNetworkInfo, err = hns.getNetworkByName(hnsNetworkName)
+	}
+	return hnsNetworkInfo, err
+}
+
+func isOverlay(hnsNetworkInfo *hnsNetworkInfo) bool {
+	return strings.EqualFold(hnsNetworkInfo.networkType, NETWORK_TYPE_OVERLAY)
+}
+
+func (svcInfo *serviceInfo) cleanupAllPolicies(endpoints []proxy.Endpoint) {
+	klog.V(3).InfoS("Service cleanup", "serviceInfo", svcInfo)
+	// Skip the svcInfo.policyApplied check to remove all the policies
+	svcInfo.deleteAllHnsLoadBalancerPolicy()
+	// Cleanup Endpoints references
+	for _, ep := range endpoints {
+		epInfo, ok := ep.(*endpointsInfo)
+		if ok {
+			epInfo.Cleanup()
+		}
+	}
+	if svcInfo.remoteEndpoint != nil {
+		svcInfo.remoteEndpoint.Cleanup()
+	}
+
+	svcInfo.policyApplied = false
+}
+
+func (svcInfo *serviceInfo) deleteAllHnsLoadBalancerPolicy() {
+	// Remove the Hns Policy corresponding to this service
+	hns := svcInfo.hns
+	hns.deleteLoadBalancer(svcInfo.hnsID)
+	svcInfo.hnsID = ""
+
+	hns.deleteLoadBalancer(svcInfo.nodePorthnsID)
+	svcInfo.nodePorthnsID = ""
+
+	for _, externalIP := range svcInfo.externalIPs {
+		hns.deleteLoadBalancer(externalIP.hnsID)
+		externalIP.hnsID = ""
+	}
+	for _, lbIngressIP := range svcInfo.loadBalancerIngressIPs {
+		hns.deleteLoadBalancer(lbIngressIP.hnsID)
+		lbIngressIP.hnsID = ""
+		if lbIngressIP.healthCheckHnsID != "" {
+			hns.deleteLoadBalancer(lbIngressIP.healthCheckHnsID)
+			lbIngressIP.healthCheckHnsID = ""
+		}
+	}
+}
+
+func deleteAllHnsLoadBalancerPolicy() {
+	lbalancers, err := hcn.ListLoadBalancers()
+	if err != nil {
+		return
+	}
+	for _, lbalancer := range lbalancers {
+		klog.V(3).InfoS("Remove policy", "policies", lbalancer)
+		err = lbalancer.Delete()
+		if err != nil {
+			klog.ErrorS(err, "Failed to delete policy list")
+		}
+	}
+}
+
+func getHnsNetworkInfo(hnsNetworkName string) (*hnsNetworkInfo, error) {
```

Comment:
Is there a significant difference between this and ``getNetworkInfo``? Should we have both?